internal/host/hostcommands/install_cmd.go

```diff
@@ -391,16 +397,32 @@ func appendISCSIArgs(installerArgs []string, installationDisk *models.Disk, inve
 	if iSCSIHostIP.Is6() {
 		dhcp = "dhcp6"
 	}
-	installerArgs = append(installerArgs, "--append-karg", fmt.Sprintf("ip=%s:%s", nic.Name, dhcp))
+
+	if !slices.Contains(installerArgs, fmt.Sprintf("ip=%s:%s", nic.Name, dhcp)) {
+		installerArgs = append(installerArgs, "--append-karg", fmt.Sprintf("ip=%s:%s", nic.Name, dhcp))
+	}
 
 	return installerArgs, nil
 }
 
-func appendMultipathArgs(installerArgs []string, installationDisk *models.Disk) []string {
+func appendMultipathArgs(installerArgs []string, installationDisk *models.Disk, inventory *models.Inventory, hasUserConfiguredIP bool) ([]string, error) {
 	if installationDisk.DriveType != models.DriveTypeMultipath {
-		return installerArgs
+		return installerArgs, nil
 	}
-	return append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
+
+	installerArgs = append(installerArgs, "--append-karg", "root=/dev/disk/by-label/dm-mpath-root", "--append-karg", "rw", "--append-karg", "rd.multipath=default")
+
+	for _, disk := range inventory.Disks {
+		if disk.DriveType == models.DriveTypeISCSI && strings.Contains(disk.Holders, installationDisk.Name) {
+			iSCSIInstallerArgs, err := appendISCSIArgs(installerArgs, disk, inventory, hasUserConfiguredIP)
```

Comment:
```suggestion
			installerArgs, err := appendISCSIArgs(installerArgs, disk, inventory, hasUserConfiguredIP)
```
what is returned are all args that will be passed to the installer, not only the iSCSI ones

Reply:
but if I remove the break, each iteration needs to take the returned arguments and append them to the installerArgs. Maybe the name isn’t  accurate

Reply:
`appendISCSIArgs` function already append items to `installerArgs`